lib/util/flattenOperands.js

```diff
@@ -34,293 +35,80 @@ interested in how that works
-  const operands = getOperands(node.args[0], '*');
-
-  // If the last operand (so far) under * was a constant, then it's a
-  // polynomial term.
-  // e.g. 2*5*6x creates a tree where the top node is implicit multiplcation
-  // and the left branch goes to the tree with 2*5*6, and the right operand
-  // is the symbol x. We want to check that the last argument on the left (in
-  // this example 6) is a constant.
-  const lastOperand = operands.pop();
-
-  // in the above example, node.args[1] would be the symbol x
-  const nextOperand = flattenOperands(node.args[1]);
-
-  // a coefficient can be constant or a fraction of constants
-  if (Node.Type.isConstantOrConstantFraction(lastOperand)) {
-    // we replace the constant (which we popped) with constant*symbol
-    operands.push(
-      Node.Creator.operator('*', [lastOperand, nextOperand], true));
-  }
-  // Now we know it isn't a polynomial term, it's just another seperate operand
-  else {
-    operands.push(lastOperand);
-    operands.push(nextOperand);
-  }
-  return operands;
-}
-
-// Takes a division node and returns a list of operands
-// If there is multiplication in the numerator, the operands returned
-// are to be multiplied together. Otherwise, a list of length one with
-// just the division node is returned. getOperands might change the
-// operator accordingly.
-function flattenDivision(node) {
-  // We recurse on the left side of the tree to find operands so far
-  // Flattening division is always considered part of a bigger picture
-  // of multiplication, so we get operands with '*'
-  let operands = getOperands(node.args[0], '*');
-
-  if (operands.length === 1) {
-    node.args[0] = operands.pop();
-    node.args[1] = flattenOperands(node.args[1]);
-    operands = [node];
+function isPolynomialTerm(node) {
+  if (query.isNumber(node)) {
+    return true;
   }
-  else {
-    // This is the last operand, the term we'll want to add our division to
-    const numerator = operands.pop();
-    // This is the denominator of the current division node we're recursing on
-    const denominator = flattenOperands(node.args[1]);
-    // Note that this means 2 * 3 * 4 / 5 / 6 * 7 will flatten but keep the 4/5/6
-    // as an operand - in simplifyDivision.js this is changed to 4/(5*6)
-    const divisionNode = Node.Creator.operator('/', [numerator, denominator]);
-    operands.push(divisionNode);
+  else if (query.isIdentifier(node)) {
+    return true;
   }
-
-  return operands;
-}
-
-// Returns true if there is a * node nested in some division, with no other
-// operators or parentheses between them.
-// e.g. returns true: 2*3/4, 2 / 5 / 6 * 7 / 8
-// e.g. returns false: 3/4/5, ((3*2) - 5) / 7, (2*5)/6
-function hasMultiplicationBesideDivision(node) {
-  if (!Node.Type.isOperator(node)) {
-    return false;
+  else if (query.isPow(node)) {
+    const [base, exponent] = node.args;
+    return query.isIdentifier(base) && isPolynomialTerm(exponent);
```

Comment:
I figured it would be nice to be able to describe things like `x^a * x^b --> x^(a+b)` where `a` and `b` are actually `a` and `b`.